scripts/core/EXP-COMMAND-BUFFER.rst

```diff
@@ -147,6 +158,82 @@ enqueued or executed simultaneously, and submissions may be serialized.
     ${x}CommandBufferEnqueueExp(hCommandBuffer, hQueue, 0, nullptr,
                               &executionEvent);
 
+Updating Command-Buffers Commands
+--------------------------------------------------------------------------------
+
+An adapter implementing the command-buffer experimental feature can optionally
+support updating the configuration of kernel commands recorded to a
+command-buffer. Support for this is reported by returning true in the
+${X}_DEVICE_INFO_COMMAND_BUFFER_UPDATE_SUPPORT_EXP query.
+
+Updating kernel commands is done by passing the new kernel configuration
+to ${x}CommandBufferUpdateKernelLaunchExp along with the command handle of
+the kernel command to update. Configurations that can be changed are the
+parameters to the kernel and the execution ND-Range.
+
+.. parsed-literal::
+
+    // Create a command-buffer with update enabled.
+    ${x}_exp_command_buffer_desc_t desc {
+      ${X}_STRUCTURE_TYPE_EXP_COMMAND_BUFFER_DESC,
+      nullptr,
+      true // isUpdatable
+    };
+    ${x}_exp_command_buffer_handle_t hCommandBuffer;
+    ${x}CommandBufferCreateExp(hContext, hDevice, &desc, &hCommandBuffer);
+
+    // Append a kernel command which has two buffer parameters, an input
+    // and an output.
+    ${x}_exp_command_buffer_command_handle_t hCommand;
+    ${x}CommandBufferAppendKernelLaunchExp(hCommandBuffer, hKernel, workDim,
+                                           pGlobalWorkOffset, pGlobalWorkSize,
+                                           pLocalWorkSize, 0, nullptr,
+                                           nullptr, &hCommand);
+
+    // Close the command-buffer before updating
+    ${x}CommandBufferFinalizeExp(hCommandBuffer);
```

Comment:
This implies to me that finalizing is required before updating, and if that is the intention (it is not reflected in errors returned in the YAML at least), do we need to mandate that the command buffer be finalized before updating? I think it is possible to update a non-executable CUDA Graph as well as an executable one at least.

Reply:
Good question, in [clUpdateMutableCommandsKHR](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_Ext.html#mutable-commands) for the OpenCL backend there is the error:
> CL_INVALID_OPERATION if command_buffer has not been finalized.

So I think for version 1 of this feature we should go with that restriction, then we could look into relaxing it if we need to. I'll update the docs to reflect this.